apps/dashboard/src/main/java/com/akto/action/observe/InventoryAction.java

```diff
@@ -416,25 +418,56 @@ public String fetchSensitiveParamsForEndpoints() {
         return Action.SUCCESS.toUpperCase();
     }
 
-    public String loadRecentApiInfos(){
-        Bson filter = Filters.and(
-            Filters.nin(ApiInfo.ID_API_COLLECTION_ID,deactivatedCollections),
-            Filters.gte(ApiInfo.DISCOVERED_TIMESTAMP, startTimestamp),
-            Filters.lte(ApiInfo.DISCOVERED_TIMESTAMP, endTimestamp)
-        );
-        List<ApiInfo> apiInfos = ApiInfoDao.instance.findAll(filter);
-        for(ApiInfo info: apiInfos){
-            info.calculateActualAuth();
-        }
+    public String getSummaryInfoForChanges(){
+        long countEndpoints = SingleTypeInfoDao.instance.fetchEndpointsCount(startTimestamp, endTimestamp, deactivatedCollections);
+
+        List<String> sensitiveSubtypes = new ArrayList<>();
+        sensitiveSubtypes.addAll(SingleTypeInfoDao.instance.sensitiveSubTypeInResponseNames());
+        sensitiveSubtypes.addAll(SingleTypeInfoDao.instance.sensitiveSubTypeInRequestNames());
+        int countSensitiveApis = SingleTypeInfoDao.instance.getSensitiveApisCount(sensitiveSubtypes, false, (
+            Filters.and(
+                    Filters.and(
+                        Filters.gte(SingleTypeInfo._TIMESTAMP, startTimestamp),
+                        Filters.lte(SingleTypeInfo._TIMESTAMP, endTimestamp)
+                    ),
+                    Filters.nin(SingleTypeInfo._API_COLLECTION_ID, deactivatedCollections)
+            )
+        ));
+
         response = new BasicDBObject();
-        response.put("apiInfoList", apiInfos);
+        response.put("newEndpointsCount", countEndpoints);
+        response.put("sensitiveEndpointsCount", countSensitiveApis);
+
         return Action.SUCCESS.toUpperCase();
     }
 
     public String loadRecentEndpoints() {
-        List<BasicDBObject> list = fetchRecentEndpoints(startTimestamp, endTimestamp);
+        String regexPattern = getRegexPattern();
+        Bson searchFilter = Filters.empty();
+        if(!regexPattern.isEmpty() && regexPattern.length() > 0){
+            searchFilter = Filters.or(
+                Filters.regex(ApiInfo.ID_METHOD, regexPattern, "i"),
+                Filters.regex(ApiInfo.ID_URL, regexPattern, "i")
+            );
+        }
+        if(skip < 0){
+            skip *= -1;
+        }
```

Comment:
how does negative skip work here, if the skip becomes less than available docs, and in general